src/main/java/seedu/address/model/reminder/Reminder.java

```diff
@@ -0,0 +1,117 @@
+package seedu.address.model.reminder;
+
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.lead.Lead;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a Reminder
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Reminder {
+    private static final int DAYS_IN_WEEK = 7;
+
+    // Identity fields
+    private final Name name;
+    private final Phone phone;
+
+    // Data fields
+    private final Set<Tag> tags = new HashSet<>();
+    private final Lead lead;
+
+    private final Date currentDate;
+    private final Date followUpDate;
+
+    /**
+     * Represents a Reminder in the list of reminders
+     * Guarantees: details are present and not null, field values are validated, immutable.
+     */
+    public Reminder(Name name, Phone phone, Set<Tag> tags, Lead lead, Date currentDate) {
+        this.name = name;
+        this.phone = phone;
+        this.tags.addAll(tags);
+        this.lead = lead;
+        this.currentDate = currentDate;
+        this.followUpDate = this.getFollowUpDate(currentDate, lead);
+    }
+
+    public Name getName() {
+        return name;
+    }
+
+    public Phone getPhone() {
+        return phone;
+    }
+
+    /**
+     * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
+    public Set<Tag> getTags() {
+        return Collections.unmodifiableSet(tags);
+    }
+
+    public Lead getLead() {
+        return lead;
+    }
+
+    public Date getCurrentDate() {
+        return currentDate;
+    }
+
+    public Date getFollowUpDate() {
+        return followUpDate;
+    }
```

Comment:
I think having lots of getters like this might be rather unhealthy. Perhaps some of the filtering logic can be moved to be done internally by `Reminder` rather than the `UniqueReminderList`?

Reply:
For the getters I put them there more for future use to get the info for the javafx components, but comparison between Reminder and date can definitely be done within Reminder